# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/remotebrowserutils/RemoteWebNavigation.jsm
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc toolkit/components/remotebrowserutils/RemoteWebNavigation.jsm
--- a/toolkit/components/remotebrowserutils/RemoteWebNavigation.jsm	Wed Apr 28 11:09:38 2021 +0000
+++ b/toolkit/components/remotebrowserutils/RemoteWebNavigation.jsm	Wed Apr 28 11:26:49 2021 +0000
@@ -59,7 +59,8 @@
     );
     this._browser.browsingContext.goBack(
       cancelContentJSEpoch,
-      requireUserInteraction
+      requireUserInteraction,
+      true
     );
   }
   goForward(requireUserInteraction = false) {
@@ -68,7 +69,8 @@
     );
     this._browser.browsingContext.goForward(
       cancelContentJSEpoch,
-      requireUserInteraction
+      requireUserInteraction,
+      true
     );
   }
   gotoIndex(aIndex) {
@@ -76,7 +78,7 @@
       Ci.nsIRemoteTab.NAVIGATE_INDEX,
       { index: aIndex }
     );
-    this._browser.browsingContext.goToIndex(aIndex, cancelContentJSEpoch);
+    this._browser.browsingContext.goToIndex(aIndex, cancelContentJSEpoch, true);
   }
   loadURI(aURI, aLoadURIOptions) {
     let uri;
